docs/application/web/api/6.5/device_api/mobile/tizen/ml_pipeline.html

```diff
@@ -0,0 +1,2120 @@
+          </div>
+<pre class="webidl prettyprint">  [Callback=FunctionOnly, NoInterfaceObject] interface SinkListener {
+    void ondata(DOMString sinkName, <a href="ml.html#TensorsData">TensorsData</a> data);
+  };</pre>
+<p><span class="version">Since: </span>
+ 6.5
+          </p>
+<div class="methods">
+<h4>Methods</h4>
+<dl>
+<dt class="method" id="SinkListener::ondata">
+<a class="backward-compatibility-anchor" name="::Pipeline::SinkListener::ondata"></a><code><b><span class="methodName">ondata</span></b></code>
+</dt>
+<dd>
+<div class="brief">
+ Called when new data arrives to the sink.
+            </div>
+<div class="synopsis"><pre class="signature prettyprint">void ondata(DOMString sinkName, <a href="ml.html#TensorsData">TensorsData</a> data);</pre></div>
+<p><span class="version">Since: </span>
+ 6.5
+            </p>
+<div class="parameters">
+<p><span class="param">Parameters:</span></p>
+<ul>
+          <li class="param">
+<span class="name">sinkName</span>:
+ Name of the sink.
+                </li>
+          <li class="param">
+<span class="name">data</span>:
+ The data that arrived to the sink.
+                </li>
+        </ul>
+</div>
+</dd>
+</dl>
+</div>
+</div>
+<div class="interface" id="CustomFilter">
+<a class="backward-compatibility-anchor" name="::Pipeline::CustomFilter"></a><h3>2.9. CustomFilter</h3>
+<div class="brief">
+ The <em>CustomFilter</em> interface defines the callback registered with <a href="#MachineLearningPipeline::registerCustomFilter">MachineLearningPipeline::registerCustomFilter()</a>.
+          </div>
+<pre class="webidl prettyprint">  [Callback=FunctionOnly, NoInterfaceObject] interface CustomFilter {
+    long filter(<a href="ml.html#TensorsData">TensorsData</a> input, <a href="ml.html#TensorsData">TensorsData</a> output);
+  };</pre>
+<p><span class="version">Since: </span>
+ 6.5
+          </p>
+<div class="methods">
+<h4>Methods</h4>
+<dl>
+<dt class="method" id="CustomFilter::filter">
+<a class="backward-compatibility-anchor" name="::Pipeline::CustomFilter::filter"></a><code><b><span class="methodName">filter</span></b></code>
+</dt>
+<dd>
+<div class="brief">
+ Called when data to be processed arrives to the filter.
+            </div>
+<div class="synopsis"><pre class="signature prettyprint">long filter(<a href="ml.html#TensorsData">TensorsData</a> input, <a href="ml.html#TensorsData">TensorsData</a> output);</pre></div>
+<p><span class="version">Since: </span>
+ 6.5
+            </p>
+<div class="description">
+            <p>
+This user-defined callback should transform the <em>input</em> and write the transformation
+result to <em>output</em>. <em>output</em>'s contents will be passed to the further stages of the pipeline.
+            </p>
+           </div>
+<p><span class="remark">Remark: </span>
+ <em>input</em> and <em>output</em> are only valid within the callback. They are automatically disposed after callback's return.
```

Comment:
The<em>input</em> and the <em>output</em> are only valid within the callback. They are automatically disposed after callback's return. 

What is this 
Copy them to access their contents outside of the callback.

Reply:
What do you think about the new version below?

```suggestion
The <em>input</em> and the <em>output</em> are only valid within the callback. They are automatically disposed after callback's return.
 To use their contents after callback return, copy their contents to objects valid outside the callback.
```